packages/extension-base/src/background/handlers/Extension.ts

```diff
@@ -333,6 +338,19 @@ export default class Extension {
       pair.decodePkcs8(password);
     }
 
+    if (isJsonPayload(request.payload)) {
+      // Get the metadata for the genesisHash
+      const currentMetadata = this.#state.knownMetadata.find((meta: MetadataDef) =>
+        meta.genesisHash === (request.payload as SignerPayloadJSON).genesisHash);
```

Comment:
```suggestion
        meta.genesisHash === request.payload.genesisHash;
```

The guard above ensures the value is what it is meant to be. (If TS does complain, which it shouldn't however since it is nsted, then `const { request } = payload` so it is stand-alone would nudge it in the right direction before the guard)

Reply:
yes.. it did complain.. thanks for your suggestion.. made this change..